### ISSUE TITLE
[Bug]: "TypeError: request.formData is not a function" in the action handler in unit tests

### ISSUE THREAD

**Issue by duttanuj** [remix-run/react-router#10074]
### What version of React Router are you using?

6.4.3

### Steps to Reproduce

Hi! I've been working on a project where I've used the `useFetcher` hook to submit a request to the backend. I'm trying to add a unit test where I can mock the `fetch` call using `jest-fetch-mock` and then assert on the way my component handles various backend responses. Here's a rough code example of what I'm trying to do:

```
function UseFetcherComponent() {

  const fetcher = useFetcher();

  const handleFormSubmit = (event: FormEvent<HTMLFormElement>) => {
    event.preventDefault();

    .. some validation etc

    fetcher.submit(event.currentTarget, {
      method: 'post',
    });
  };

  return (
    <fetcher.Form onSubmit={handleFormSubmit}>
          <Form actions={<Button formAction="submit">Submit</Button>}>
            .. some form components
          </Form>
    </fetcher.Form>
  )
}
```

In my test I'm trying to mock the action handler like this: 

```
async function testAction({ request }: ActionFunctionArgs) {
  const formData = await request.formData();
  return dispatchBackendRequest(formData);
}

function renderWithAction() {
  const routes: RouteObject[] = [
    {
      path: '/',
      element: <UseFetcherComponent />,
      action: testAction,
    },
  ];
  const router = createMemoryRouter(routes, {
    initialEntries: ['/'],
  });

  render(<RouterProvider router={router} />);
  return wrapper(document.documentElement);
}

describe('<UseFetcherComponent />', () => {
  it('calls the backend', async () => {
    const testContainer = renderWithAction();
    // Use selectors to click the submit button
    expect(fetchMock).toBeCalled();
  });
});
```

### Expected Behavior

Expected fetchMock to be called. This code works normally. I expect it to run through the motions in the tests too. 

### Actual Behavior

I get an error saying `TypeError: request.formData is not a function`.

Basically the testAction breaks:

```
async function testAction({ request }: ActionFunctionArgs) {
  const formData = await request.formData();  // <---- Breaks!
  return dispatchBackendRequest(formData);
}

```

I tried to console.log the request object received and here's the log entry:

```
{
  "size": 0,
  "timeout": 0,
  "follow": 20,
  "compress": true,
  "counter": 0
}
```

What am I missing?

**Comment by seanmcquaid** [comment]
Seeing the same thing!

**Comment by ktj** [comment]
node doesn't have FormData, maybe. that has something to do with this?
Without looking react router code, could a polyfill fix this?

**Comment by timdorr** [comment]
This seems like a limitation of jsdom, not our library.

**Comment by ktj** [comment]
I upgraded to node18 and removed fetch polyfill, it resolved this.
